chain/src/txhashset/txhashset.rs

```diff
@@ -1242,8 +1242,9 @@ impl<'a> Extension<'a> {
 	/// from the respective MMRs.
 	/// For a significantly faster way of validating full kernel sums see BlockSums.
 	pub fn validate_kernel_sums(&self) -> Result<((Commitment, Commitment)), Error> {
+		let genesis = self.get_header_by_height(0)?;
 		let (utxo_sum, kernel_sum) = self.verify_kernel_sums(
-			self.header.total_overage(),
+			self.header.total_overage(genesis.kernel_mmr_size > 0),
```

Comment:
This should probably be based on chain type eventually?

Reply:
That was my first inkling as well but it makes it a pain for tests. Tests should all be running in AutomatedTesting mode but some use an empty genesis while others won't.